ion/integration_test.go

```diff
@@ -346,19 +346,10 @@ func testBinaryRoundTrip(t *testing.T, fp string) {
 	// Re-encode binWriter's stream as text into a string builder
 	str := encodeAsTextIon(t, buf.String())
 
-	reader1 := NewReader(bytes.NewReader(buf.Bytes()))
-	reader2 := NewReader(strings.NewReader(str.String()))
-
-	for reader1.Next() {
-		i1 := readCurrentValue(t, reader1)
-		reader2.Next()
-		i2 := readCurrentValue(t, reader2)
-
-		if !reflect.DeepEqual(i1.value, i2.value) ||
-			!reflect.DeepEqual(i1.annotations, i2.annotations) ||
-			!reflect.DeepEqual(i1.ionType, i2.ionType) {
-			t.Errorf("failed %s %v vs %v", fp, i1.value, i2.value)
-		}
+	buf2 := encodeAsBinaryIon(t, []byte(str.String()))
+
+	if !reflect.DeepEqual(buf, buf2) {
```

Comment:
We don't want to compare the buffers or strings that are produced directly; the encoded bytes can be different but still contain equivalent values. For example, the `TextWriter` and the `PrettyTextWriter` will produce different `string`s, but the values in the strings should be considered the same. Similarly, some data types like `Timestamp` have a variety of ways to encode the same value; `2007-02-23T00:00Z` and `2007-02-23T00:00+00:00` are equivalent despite being encoded differently.

Reply:
I was hoping you read the email I sent about this before seeing it in the PR :P
Updated the test as talked yesterday.